src/Bongo/Actors/BulkOperations/BulkInsertWorker.cs

```diff
@@ -129,9 +140,10 @@ private List<AddRangePartitionRequest> CheckForNeededPartitions(InsertRequest re
                     switch (value)
                     {
                         case DateTimeOffset time: return time.ToUnixTimeMilliseconds();
+                        case DateTime time: return new DateTimeOffset(time).ToUnixTimeMilliseconds();
                         case int time: return time;
                         case long time: return time;
-                        default: throw new Exception("cannot parse");
+                        default: throw new Exception($"cannot get the partition value for type '{value.GetType().Name}'");
```

Comment:
Should the message mention that the particular type is invalid or not supported... and maybe list the valid types?